examples/grpc-server-with-prometheus/server/server.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	pb "github.com/grpc-ecosystem/go-grpc-prometheus/examples/grpc-server-with-prometheus/protobuf"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// DemoServiceServer defines a Server.
+type DemoServiceServer struct{}
+
+func newDemoServer() *DemoServiceServer {
+	return &DemoServiceServer{}
+}
+
+// SayHello implements a interface defined by protobuf.
+func (s *DemoServiceServer) SayHello(ctx context.Context, request *pb.HelloRequest) (*pb.HelloResponse, error) {
+	pushCounter.Inc()
+	return &pb.HelloResponse{Message: fmt.Sprintf("Hello %s", request.Name)}, nil
+}
+
+// Create a counter metric
+var (
+	pushCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "repository_pushes",
+		Help: "Number of pushes to external repository.",
+	})
+)
+
+func init() {
+	// Metrics have to be registered to be exposed:
+	prometheus.MustRegister(pushCounter)
+}
+
+func main() {
+	// Listen an actual port.
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", 9093))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+
+	// Create a HTTP server for prometheus.
+	httpServer := &http.Server{Handler: promhttp.Handler(), Addr: fmt.Sprintf("0.0.0.0:%d", 9092)}
+
+	// Create a gRPC Server with gRPC interceptor.
+	grpcServer := grpc.NewServer(
+		grpc.StreamInterceptor(grpc_prometheus.StreamServerInterceptor),
+		grpc.UnaryInterceptor(grpc_prometheus.UnaryServerInterceptor),
+	)
+
+	// Register your service.
+	pb.RegisterDemoServiceServer(grpcServer, newDemoServer())
+
+	// Enable Histogram for gRPC-Prometheus interceptor.
+	grpc_prometheus.EnableHandlingTimeHistogram()
+
+	// Register your gRPC Server to gRPC-Prometheus interceptor.
+	grpc_prometheus.Register(grpcServer)
+
+	// Start your http server for prometheus.
+	go func() {
+		if err := httpServer.ListenAndServe(); err != nil {
+			log.Fatal("Unable to start a http server.")
```

Comment:
Still graceful shutdown would be nice to add here. What do you think? (: 

Graceful shutdown means the ability to complete all the requests with given timeout on e.g interrupt (Ctrl+C)
Currently, this program probably does not care about Ctrl+C (interrupt signal) or is doing something random on that.

Reply:
I agree that this demo service should be graceful shutdown. I consider the package called run.Group. However, i propose that don't add unstandard package to this demo, it like cmux.
I'm finding a method through golang standard packages to close gracefully it. 
Do you have a suitable method to handle it  except run.Group?

Reply:
fair point. Just use some interrupt method
```
func interrupt(cancel <-chan struct{}) error {
	c := make(chan os.Signal, 1)
	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
	select {
	case <-c:
		return nil
	case <-cancel:
		return errors.New("canceled")
	}
}
```
And setup plain go routines and channels to make that work together with servers. (Stop all servers if interrupt returned)

Reply:
However without run.Groups it will be ugly and hard not to make mistake. If you don't want to use `run.Group` I think it is better to just state in comment `NOTE: Graceful shutdown is missing`. Or something like that. The truth is in this example we don't want to teach people to start server.. we want show how to setup grpc Prometheus monitoring.

Reply:
or we can think about it in next PR. I think that moving to local prometheus.Registry is more crucial

Reply:
Still graceful shutdown must be nice and correct. If those code are implemted to a real service in production, i certainly to use run.Group. As you ever said above ”The truth is in this example we don't want to teach people to start server.. we want show how to setup grpc Prometheus monitoring“， I strongly agree it. The example is as simple as possible, its target is that let beginners to utilize quickly the go-grpc-prometheus.  Even so ,  your suggesstion let me know run.Group and remind me notice the "graceful shutdown". Thanks.

Reply:
ok